packages/domains-table/src/domains-table/index.tsx

```diff
@@ -16,15 +18,52 @@ interface DomainsTableProps {
 
 export function DomainsTable( { domains, fetchSiteDomains, isAllSitesView }: DomainsTableProps ) {
 	const { __ } = useI18n();
+	const [ selectedDomains, setSelectedDomains ] = useState< PartialDomainData[] >( [] );
+
+	useLayoutEffect( () => {
+		setSelectedDomains( [] );
+	}, [ domains ] );
+
+	const handleSelectDomain = useCallback(
+		( domain: PartialDomainData ) => {
+			if ( selectedDomains.includes( domain ) ) {
+				setSelectedDomains(
+					selectedDomains.filter( ( selectedDomain ) => selectedDomain !== domain )
+				);
+			} else {
+				setSelectedDomains( [ ...selectedDomains, domain ] );
+			}
```

Comment:
Using a `Set` would make this logic easier too 😊

Reply:
Would a `WeakSet` be even better so that the dangling references are cleaned? :tinfoil-hat:

Reply:
Since I'd recommend storing `string`s in the set then there aren't issues with dangling references. That's more an issue if we continue to hold objects.